src/main/java/hm13/ThreadSafeList.java

```diff
@@ -0,0 +1,55 @@
+package hm13;
+
+import java.util.List;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+public class ThreadSafeList<T> implements Runnable{
```

Comment:
```suggestion
public class ThreadSafeList<T> {
```
Runnable також не потрібен (метод run у тебе пустий).